libraries/AP_RangeFinder/AP_RangeFinder.cpp

```diff
@@ -666,10 +666,13 @@ RangeFinder::Status RangeFinder::status_orient(enum Rotation orientation) const
 
 void RangeFinder::handle_msg(const mavlink_message_t &msg)
 {
+    mavlink_distance_sensor_t packet;
+    mavlink_msg_distance_sensor_decode(&msg, &packet);
+    
     uint8_t i;
     for (i=0; i<num_instances; i++) {
-        if ((drivers[i] != nullptr) && ((Type)params[i].type.get() != Type::NONE)) {
-          drivers[i]->handle_msg(msg);
+        if ((drivers[i] != nullptr) && (packet.orientation == drivers[i]->orientation()) && ((Type)params[i].type.get() == Type::MAVLink)) {
```

Comment:
As PeterB says we shouldn't have the orientation and type check in here.  That's really for the backend to figure out I think.

There probably also needs to be a check at the top that the message is really a distance-sensor message or we need to rename this method as well to handle_distance_sensor_msg.

Reply:
> There probably also needs to be a check at the top that the message is really a distance-sensor message or we need to rename this method as well to handle_distance_sensor_msg.

It is called by **GCS_MAVLINK::handle_distance_sensor** only so yes we need to rename it,